robolectric/src/test/java/org/robolectric/shadows/ShadowBitmapFactoryTest.java

```diff
@@ -356,9 +461,66 @@ public void decodeFileDescriptor_shouldHaveCorrectWidthAndHeight() throws IOExce
-      assertEquals(500, loadedBitmap.getWidth());
-      assertEquals(600, loadedBitmap.getHeight());
+      assertThat(loadedBitmap.getWidth()).isEqualTo(TEST_JPEG_WIDTH);
+      assertThat(loadedBitmap.getHeight()).isEqualTo(TEST_JPEG_HEIGHT);
       loadedBitmap.recycle();
     }
   }
+
+  private void decodeFile_shouldGetCorrectColorFromCompressedFile(
+      Bitmap.CompressFormat format, byte[] bitmapData) throws IOException {
+    Bitmap oldBitmap = BitmapFactory.decodeByteArray(bitmapData, 0, bitmapData.length);
+    Path tempFile = Files.createTempFile("bitmap", null);
+    FileOutputStream fileOutputStream = new FileOutputStream(tempFile.toFile());
+    // lossless compression
+    oldBitmap.compress(format, 100, fileOutputStream);
+    fileOutputStream.close();
+    Bitmap newBitmap = BitmapFactory.decodeFile(tempFile.toAbsolutePath().toString());
+
+    ByteBuffer oldBuffer = ByteBuffer.allocate(oldBitmap.getHeight() * oldBitmap.getRowBytes());
+    oldBitmap.copyPixelsToBuffer(oldBuffer);
+
+    ByteBuffer newBuffer = ByteBuffer.allocate(newBitmap.getHeight() * newBitmap.getRowBytes());
+    newBitmap.copyPixelsToBuffer(newBuffer);
+    assertThat(oldBuffer.array()).isEqualTo(newBuffer.array());
+  }
+
+  private int getPngImageColorFromStream(String pngImagePath) throws IOException {
+    InputStream inputStream =
+        new BufferedInputStream(getClass().getClassLoader().getResourceAsStream(pngImagePath));
+    inputStream.mark(inputStream.available());
+    BitmapFactory.Options opts = new BitmapFactory.Options();
+    Bitmap bitmap = BitmapFactory.decodeStream(inputStream, null, opts);
+    return bitmap.getPixel(0, 0);
+  }
+
+  private int getPngImageColorFromByteArray(String pngImagePath) throws IOException {
+    InputStream inputStream =
+        new BufferedInputStream(getClass().getClassLoader().getResourceAsStream(pngImagePath));
+    inputStream.mark(inputStream.available());
+    byte[] array = new byte[inputStream.available()];
+    inputStream.read(array);
+    Bitmap bitmap = BitmapFactory.decodeByteArray(array, 0, array.length);
+    return bitmap.getPixel(0, 0);
+  }
+
+  private int getPngImageColorFromResourceStream(String pngImagePath) throws IOException {
+    Bitmap bitmap = getBitmapFromResourceStream(pngImagePath);
+    return bitmap.getPixel(0, 0);
+  }
+
+  private Bitmap getBitmapFromResourceStream(String imagePath) throws IOException {
+    InputStream inputStream =
+        new BufferedInputStream(getClass().getClassLoader().getResourceAsStream(imagePath));
+    inputStream.mark(inputStream.available());
+    BitmapFactory.Options opts = new BitmapFactory.Options();
+    Resources resources = context.getResources();
+    return BitmapFactory.decodeResourceStream(resources, null, inputStream, null, opts);
+  }
+
+  private byte[] getBitmapByteArrayFromResourceStream(String imagePath)
+      throws IOException, URISyntaxException {
+    return Files.readAllBytes(
```

Comment:
Can you use Guava's Resources.toByteArray for this? e.g. `com.google.common.io.Resources.toByteArray(com.google.common.io.Resources.getResource(imagePath));`. For some reason when I run this test in my environment (bazel), I get a FileNotFoundException:

```
java.nio.file.FileSystemNotFoundException
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getFileSystem(ZipFileSystemProvider.java:169)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getPath(ZipFileSystemProvider.java:155)
	at java.base/java.nio.file.Path.of(Path.java:208)
	at java.base/java.nio.file.Paths.get(Paths.java:97)
	at org.robolectric.shadows.ShadowBitmapFactoryTest.getBitmapByteArrayFromResourceStream(ShadowBitmapFactoryTest.java:526)
	at org.robolectric.shadows.ShadowBitmapFactoryTest.decodeFile_shouldGetCorrectColorFromCompressedWebpFile(ShadowBitmapFactoryTest.java:430)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:57)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:59)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:575)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:278)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:835)
```